src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -419,6 +419,14 @@ protected function getEnv($name)
         if (false !== $env = getenv($name)) {
             return $this->envCache[$name] = $env;
         }
+        if (0 < $i = strpos($name, ':')) {
+            $id = substr($name, 0, $i);
+            $service = isset($this->services[$id]) ? $this->services[$id] : (isset($this->methodMap[$id]) ? $this->{$this->methodMap[$id]}() : $this->get($id));
+
+            if (null !== $env = $service->getEnv(substr($name, 1 + $i))) {
```

Comment:
Don't we need an interface for services with such ability, ensuring the `getEnv()` method?